### ISSUE TITLE
can lama generate various result for one same pair of mask and image?

### ISSUE THREAD

**Issue by universewill** [saic-mdal/lama#176]
can lama generate various result for one same pair of mask and image?

**Comment by Ellohiye** [comment]
Hello! I saw your question, have you implemented this code with a pre-trained model? Have you encountered this problem?‘


**(lama) root@dl-0916060380-pod-jupyter-2prjb:~/lama# python3 bin/predict.py model.path=../celebahqmodel indir=../testimg/celeba/img outdir=../output
Detectron v2 is not installed
[2022-11-02 17:15:36,769][saicinpainting.utils][WARNING] - Setting signal 10 handler <function print_traceback_handler at 0x7f4f2e195b70>
[2022-11-02 17:15:36,772][__main__][CRITICAL] - Prediction failed due to [Errno 2] No such file or directory: '../celebahqmodel/config1.yaml':**


Configuration file not found when loading pretrained model. Looking forward to your reply！

**Comment by cohimame** [comment]
Hello!

The answer is No, 
because lama generator has deterministic structure: it's simply a convolutional autoencoder with no noise injections.

**Comment by tommycwh** [comment]
I see that there is an option ```noise_fill_hole``` in the trainer. If I enable it would LAMA learns to generate image in a non-deterministic way?

https://github.com/saic-mdal/lama/blob/6bb704738d4e791106d8e87099d80831999901fc/saicinpainting/training/trainers/default.py#L61-L65

**Comment by windj007** [comment]
We tried that in our early experiments and it did not lead to any significant diversity: there is a strong supervised reconstruction loss at the core of the method, which penalizes any variations. For pluralistic generation, principally different methods are needed (purely adversarial training like in CoModGAN; or autoregressive MLE models like VQGAN+transformer or diffusion)

**Comment by tommycwh** [comment]
By strong supervised reconstruction loss, are you referring to the L1 loss only? I see in some of your config files that the weight given to the L1 loss on missing areas is 0. I think it means that basically L1 loss is not actually effective? I am just wondering if I disable the supervised reconstruction loss(es), could I train LAMA for pluralistic generation?
